src/Lifecycle.js

```diff
@@ -65,32 +66,33 @@ import sdk from './index';
  *     failed.
  */
 export function loadSession(opts) {
-    const fragmentQueryParams = opts.fragmentQueryParams || {};
     let enableGuest = opts.enableGuest || false;
     const guestHsUrl = opts.guestHsUrl;
     const guestIsUrl = opts.guestIsUrl;
+    const fragmentQueryParams = opts.fragmentQueryParams || {};
     const defaultDeviceDisplayName = opts.defaultDeviceDisplayName;
 
-    if (!guestHsUrl) {
-        console.warn("Cannot enable guest access: can't determine HS URL to use");
-        enableGuest = false;
-    }
-
-    if (enableGuest &&
-        fragmentQueryParams.guest_user_id &&
-        fragmentQueryParams.guest_access_token
-       ) {
-        console.log("Using guest access credentials");
-        return _doSetLoggedIn({
-            userId: fragmentQueryParams.guest_user_id,
-            accessToken: fragmentQueryParams.guest_access_token,
-            homeserverUrl: guestHsUrl,
-            identityServerUrl: guestIsUrl,
-            guest: true,
-        }, true).then(() => true);
-    }
+    return Promise.resolve().then(() => {
```

Comment:
This is quite difficult to read, any chance of making loadSession async?

Reply:
Yeah, probably better. Done, plus the restore function.